src/test/java/no/unit/nva/doi/FetchDoiMetadataTest.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.doi;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.MalformedParametersException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+
+import static org.junit.Assert.*;
+
+public class FetchDoiMetadataTest {
+
+    private transient FetchDoiMetadata fetchDoiMetadata;
+
+    @Before
+    public void setUp() {
+        fetchDoiMetadata = new FetchDoiMetadata();
+    }
+
+    @Test
+    public void successfulResponse() {
+        String url = "https://doi.org/10.1093/afraf/ady029";
```

Comment:
Are we requesting a real DOI? Mock

Reply:
now I mocked everything. But we did go down in code coverage to 91% (method)